File: server/games/pong/pong_modifier_base.py
from ..modifier_base import ModifierBase


class PongModifierBase(ModifierBase):

    def __init__(self):
        super().__init__()

    def on_paddle_bounce(self, game, player_id=-1):
        """Called when the ball bounces off a paddle."""
        pass

    def on_wall_bounce(self, game, player_id=-1):
        """Called when the ball bounces off a wall."""
        pass

    def on_goal(self, game, player_id=-1):
        """Called when a goal is scored."""
        pass

    def on_player_elimination(self, game, player_id):
        """Called when a player is eliminated"""
        pass

File: app/services/games/pong/pongModifierBase.ts
import { ModifierBase } from "../modifierBase";
import { Pong } from "./pong";

export class PongModifierBase extends ModifierBase {
  onGoal(game: Pong, args: { playerId: number }): void {}
  onPaddleBounce(
    game: Pong,
    args: { ballId: number; playerId: number },
  ): void {}
  onWallBounce(game: Pong, args: { wallID: number }): void {}
  onPlayerElimination(game: Pong, args: { playerId: number }): void {}
  onBallReset(game: Pong): void {}
}
